src/s3ql/fsck.py

```diff
@@ -927,16 +927,26 @@ def check_objects_temp(self):
         if not isinstance(plain_backend, LocalBackend):
             return
 
-        log.info('Checking for temporary objects (backend)...')
+        log.info('Checking for temporary objects and empty directories (backend)...')
 
-        for (path, dirnames, filenames) in os.walk(plain_backend.prefix, topdown=True):
+        empty_dirs = 0
+        for (path, dirnames, filenames) in os.walk(plain_backend.prefix, topdown=False):
             for name in filenames:
                 if not re.search(r'^[^#]+#[0-9]+--?[0-9]+\.tmp$', name):
                     continue
 
                 self.found_errors = True
                 self.log_error("removing temporary file %s", name)
                 os.unlink(os.path.join(path, name))
+                
+            if path != plain_backend.prefix:
+                try:
+                    empty_dirs += 1
```

Comment:
Why not move the counter increase after the rmdir? That way you don't need to reduce it again if there's an exception.

Reply:
Huh... I was so fixed on catching the exception, it just plain didn't occur to me that of course the rest of the try block will run if there's no exception 8-).  Definitely will do that.

Reply:
Done!